### DISCUSSION TITLE
[Routing] Catch all routes

### DISCUSSION THREAD

**Discussion by lvndry** [remix-run/remix#2365] [Proposals]
Next JS has a nice feature where you can catch all the routes starting from the slug ([documentation](https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes)).

It could be implemented like:

## Example
```
├── foo
│     ├── [...bar].js
├── foo.js
```

So:
- `/foo` would render `foo.js`. 
- `/foo/a` would render `[...bar].js`
- `/foo/a/b` would render `[...bar].js`
- `/foo/b/c/d` would render `[...bar].js`

**Comment by lvndry** [comment]
Just noticed it's already integrated in remix, but it has to be configured in the config file: https://remix.run/docs/en/v1/api/conventions#remixconfigjs

**Reply by kiliman** [reply]
You just need to define a *splat* route `$.tsx`. You access the slug with `params['*']`
```ts
// routes/blog/$.tsx
export const loader: LoaderFunction = async ({params}) => {
  const slug = params['*']
  ...
}
```
https://remix.run/docs/en/v1/api/conventions#splat-routes
